qiskit/extensions/quantum_initializer/ucg.py

```diff
@@ -47,9 +47,10 @@
 from qiskit.circuit.quantumregister import QuantumRegister
 from qiskit.circuit.quantumcircuit import QuantumCircuit
 from qiskit.exceptions import QiskitError
-from qiskit.quantum_info.synthesis import euler_angles_1q
+from qiskit.quantum_info.synthesis import OneQubitEulerDecomposer
 
 _EPS = 1e-10  # global variable used to chop very small numbers to zero
+_DECOMP1Q = OneQubitEulerDecomposer('U3')
```

Comment:
Can these be class properties? Maybe initially set to None and populated the first time the code path is reached? If the imports are made at runtime, this might resolve the circular dependency too.